### ISSUE TITLE
Forbid single if in for loop

### ISSUE THREAD

**Issue by orsinium** [wemake-services/wemake-python-styleguide#1030]
# Rule request

## Thesis

Let's start right from an example:

```python
# bad
for ... in ...:
  if ...:
    ...

# good
for ... in ...:
  if ...:
    continue
  ...
```

'if-else' and if with content before or after are ok:

```python
for ... in ...:
  if ...:
    ...
  else:
    ...

for ... in ...:
  if ...:
    ...
  ...

for ... in ...:
  ...
  if ...:
    ...
```

## Reasoning

Writing one of the rules for writing readable code is fighting with a lot of indentations. Readable code is flat code, "flat is better than nested" (PEP-20). This is one of the rules how to make code flatter.

See [Should I return from a function early or use an if statement?](https://softwareengineering.stackexchange.com/questions/18454/should-i-return-from-a-function-early-or-use-an-if-statement) for context and discussion. Probably, we should do something similar for return too, but it requires a better investigation.

**Comment by sobolevn** [comment]
Thanks a lot! I will think about this one. And inspect our own `if` usage patterns.

**Comment by sobolevn** [comment]
It think that 

```python
for ... in ...:
  if ...:
    ...
```

is also 

```python
[
  ...
  for ... in ...
  if ...
]
```

**Comment by orsinium** [comment]
Only if `...` means only one line. This issue is about a more common thing, no matter how many lines you have inside.

**Comment by sobolevn** [comment]
After noticing our internal usage patters, I can say that we use both ways. And I am not sure how can I formulate this rule to cover all cases.

**Comment by orsinium** [comment]
Can you adjust your internal patterns? :upside_down_face:

**Comment by sobolevn** [comment]
I am still researching it. I guess, it would be easier when rule prototype will be ready.
Currently I have too find them manually.

**Comment by sobolevn** [comment]
We use it quite a lot. And sometimes `if` is better than `if not` + `continue`. So, I am going to gently reject this rule.

Thanks a lot for your idea!
